documentation/source/reference.rst

```diff
@@ -0,0 +1,189 @@
+Json library reference
+**********************
+
+.. current-library:: json
+.. current-module:: json
+
+The json module
+===============
+
+Constants
+---------
+
+.. constant:: $null
+
+   Is what "null" parses to.
+
+Conditions
+----------
+
+.. class:: <json-error>
+   :open:
+   :instantiable:
+
+   All JSON errors are subclasses of this class.
+
+   :superclasses: :class:`<format-string-condition>` :drm:`<error>`
+
+.. class:: <json-parse-error>
+   :instantiable:
+
+   Any error signalled during parsing (except for file system errors)
+   will be an instance of this.
+
+   :superclasses: :class:`<json-error>`
+
+``parse-json``
+--------------
+
+.. generic-function:: parse-json
+   :open:
+
+   Parse json formatted text from the given *source*.  This is the
+   main user-visible entry point for parsing.  *table-class*, if
+   provided, should be a subclass of :class:`<table>` to use when
+   creating a json "object".
+
+   :signature: parse-json (source, #key strict?, table-class) => (json)
+   :parameter source: An :drm:`<object>`.
+   :parameter #key strict?: An instance of :drm:`<boolean>`.
+   :parameter #key table-class: Default to :class:`<string-table>`.
+   :value json: A JSON :drm:`<object>`
+
+   :discussion:
+
+      The parse is strict by default. If ``strict?:`` :drm:`#f` is
+      used then:
+
+      - `#` is allowed as a comment character
+
+      - "\<c>" is equivalent to ``<c>``, where <c> is not a defined
+        escape character.
+
+      - trailing commas are allowed in arrays and objects
+
+.. method:: parse-json
+   :specializer: <string>
+
+   Parse a JSON object from a :drm:`<string>`.
+
+   :signature: parse-json (source, #key strict?, table-class) => (json)
+   :parameter source: An instance of :drm:`<string>`
+   :parameter #key strict?: An instance of :drm:`<boolean>`. Default to :drm:`#t`.
+   :parameter #key table-class: A subclass of :class:`<table>`.
+   :value json: An instance of :drm:`<object>`.
+
+   :example:
+
+      .. code-block:: dylan
+
+        let data = """{"a": 1, "b": 2,}""";
+        let parsed = parse-json(data);
```

Comment:
I think this needs `strict?: #f` to work.

Reply:
Fixed.